### PR TITLE
rec: Prep for rec-4.5.0-alpha1

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

To make the changelist a bit shorter, I skipped doc related stuff, build related stuff and PRs that do not cause an  change in behaviour visible for users.

https://blog.powerdns.com/?p=3285&preview=true
https://builder.powerdns.com/#/builders/97/builds/1623
https://builder.powerdns.com/#/builders/114/builds/934

tag target is f0abe7ad8e6695212168119eee9186c1fd27cdf7

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    :tickets: 440
+
+    Introduce "Refresh almost expired" a mechanism to keep the record cache warm.
+
+
+  .. change::
+    :tags: Bug Fixes
+    :pullreq: 9917
+
+    Set the Protocol Buffer 'inBytes' field for zero-length messages.
```
Comment (Habbie) [inline pdns/recursordist/docs/changelog/4.5.rst:20]: This is part of the protozero change, just a week late - you can just add 9917 below or leave this one out entirely. Compared to 4.4.0 there is no difference.

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    :tags: Improvements
+    :pullreq: 9856
+
+    Use a short-lived NSEC3 hashes cache for denial validation.
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 9828
+
+    Add a compatibility layer for 'Bogus', document changes and the new metrics.
```
Comment (Habbie) [inline pdns/recursordist/docs/changelog/4.5.rst:53]: perhaps combine this with 9806 below

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 9633
+
+    two OpenBSD improvements for UDP sockets: port randomization and EAGAIN errors.
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 9630
```
Comment (Habbie) [inline pdns/recursordist/docs/changelog/4.5.rst:112]: combine with 9843 above

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    :tickets: 9621
+
+    Lookup DS entries before CNAME entries.
+
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 9841
+
+    ProtobufLogger: Better handling of short-reads, errors.
```
Comment (rgacogne) [inline pdns/recursordist/docs/changelog/4.5.rst:41]: I don't think that change is included in the recursor?

Comment (Habbie) [reply pdns/recursordist/docs/changelog/4.5.rst:41]: you're right

**Pull request merged**